src/ray/raylet/runtime_env_agent_client.cc

```diff
@@ -41,90 +43,172 @@ namespace {
 
 //------------------------------------------------------------------------------
 // Simple class to make a async POST call.
-// Will call callback exactly once with pair{non-ok, any} or pair{ok, reply body}.
+// Will call either succ_callback or fail_callback exactly once.
+//
+// Keep-alive and reuses TCP connection, but only handle 1 request at a time (no
```

Comment:
Isn't this causing any performance related issues? E.g., if one of runtime env takes long time to create, everything else will be delayed?